src/wrapper/SerializationWrapper.cpp

```diff
@@ -67,6 +67,70 @@ char fs_serializer_openmode(void* serializer)
     return ' ';
 }
 
+int fs_serializer_metainfo(void* serializer)
+{
+    return reinterpret_cast<const Serializer*>(serializer)->globalMetainfo().size();
+}
+
+void fs_serializer_metainfo_key_lengths(void* serializer, int* lengths)
```

Comment:
You are right, I didnt realize. It feels scary how easy we can break this code from user code... 
I just realized that maybe we could have different interfaces for fortran and python in order to provide something safe. For fortran we could protect it creating fortran interfaces that pass to the C counterpart the size of the array (hidden to user), and for python use boost python or something else... But you are right, that changes the design so off-topic for this PR...